### ISSUE TITLE
Swift Evolution Dashboard filter status string improvements

### ISSUE THREAD

**Issue by dempseyatgithub** [apple/swift-org-website#290]
This issue is a follow-on from PR #284 

Currently when the proposal status filter is turned on, a label appears next to it with the static text "Filtered By:" and a link that either lists the filtered proposal statuses or if there are too many of the chosen, the number of filters.

The upcoming filter flag filter added in PR #284 lists the filter status in the list header.

This issue is to:
1. Move the status string of the status/version filter to the list header
2. Make it more clear that the link next to the status filter button is used to edit the filter

The first part will put both filter status strings in a consistent place in the interface. It also gives the status/version filter more room to present its status string, allowing for more informative status.

The second part makes it more clear how to edit the status/version filter.

@kaishin @dempseyatgithub

**Comment by kaishin** [comment]
@dempseyatgithub Thank you for opening this enhancement issue.

> One approach I was thinking about was listing the status in second line subhead, which would give a good amount of space. We could possible also include the version filters, since there would be a lot more breathing room.

I like this solution. I would also move the UFO link and wording to the subhead level for a but more visual balance. We could make the actual status label either bold or emphasized to make it more scannable.

For the link next to the status filter, I am leaning towards solely using the number of filters as a toggle anchor, without an explicit edit link. If we need to spell out the word "Edit", then something like "Edit filters (1)" would also work.

**Comment by dempseyatgithub** [comment]
> @dempseyatgithub Thank you for opening this enhancement issue.

You're welcome. I didn't want this to fall through the cracks.

> > One approach I was thinking about was listing the status in second line subhead, which would give a good amount of space. We could possible also include the version filters, since there would be a lot more breathing room.
> 
> I like this solution. I would also move the UFF link and wording to the subhead level for a bit more visual balance. We could make the actual status label either bold or emphasized to make it more scannable.

One of my thoughts in keeping the two on separate lines is to avoid the grammatical gnarliness of trying to combine them into a single statement. In the mockup below, I've kept the UFF status on the first line, but de-emphasized it by reducing the font weight.

The other thought is that since there are a lot of status filters that can be set, it would be good to give that status as much room as possible.


> For the link next to the status filter, I am leaning towards solely using the number of filters as a toggle anchor, without having a separate "Edit" link. If we need to spell out the word "Edit", then something like "Edit filters (1)" would also work.

In the mockups below, when filters are set, I used "1 filter" and "n filters" as the link text. When no filter is set, I used "Edit Filters" to give the user a stronger hint of what to do. "Set Filters" might also work. I thought that would give more useful information than "No Filters" or "Zero Filters".

Note that in the last mockup I include the Swift version in the status string. The current page never shows that as part of the status string link, but with the extra space, I think it would be a nice to have. I was thinking if a range of versions was selected we could probably present "Implemented in Swift 5.1 - 5.7" or the like.

I have yet to try the pathological case of every status turned on (or every status and every version). One the one hand, we could do something like "Accepted, Implemented, and 4 Others". But there's also something appealing about the precise status if the subhead wraps to another line.

## Status filter toggled on - No filters set:

![no-filters-set](https://user-images.githubusercontent.com/470139/234868048-a9c4cba0-8b0a-484b-9277-cf153f2cdbfd.jpg)

## Filtering on three statuses:

![multiple-status-set](https://user-images.githubusercontent.com/470139/234868078-555d773c-0c4d-41b6-86d8-98f6ef7569d4.png)

## Status and UFF filters both toggled on - filtering on one status and one version:
 
![uff-and-status-filter-set](https://user-images.githubusercontent.com/470139/234868120-a86445a5-540e-4bf9-bd4d-fe2202462d2c.png)

**Comment by kaishin** [comment]
> The other thought is that since there are a lot of status filters that can be set, it would be good to give that status as much room as possible.
> I have yet to try the pathological case of every status turned on (or every status and every version). One the one hand, we could do something like "Accepted, Implemented, and 4 Others". But there's also something appealing about the precise status if the subhead wraps to another line.

Fair enough. It should be safe to display _everything_ and wrap around when needed. The total number of filters is already provided in the toggle anchor so it adds little to display it again in the heading. 

I like the Swift version range idea but it might be more implementation trouble than it's worth. There will be cases where the range isn't continuous which would require extra logic to handle and space to display.

**Comment by dempseyatgithub** [comment]
> Fair enough. It should be safe to display _everything_ and wrap around when needed. The total number of filters is already provided in the toggle anchor so it adds little to display it again in the heading.

Yes, agreed. I was just playing around with a mockup with a large number of statuses selected.

If the user selects all of them, the status can just be "All Statuses"

If the user selects all but a threshold number (Maybe 3?) we could display "All Statuses Except Returned, Withdrawn, and Rejected". I don't think the logic for that should be too convoluted.

I think we in the status string we should use the short names "Returned" and "Scheduled" that are also used in the status capsules next to the proposal. And I'd like to use "In Active Review" in the status string since 'Active Review' is the only status that doesn't fit the sentence "This proposal is <blank>".
 
> I like the Swift version range idea but it might be more implementation trouble than it's worth. There will be cases where the range isn't continuous which would require extra logic to handle and space to display.

Yes, for me listing the versions is a nice to have. It also complicates "All Statuses" and "All Statuses Except…" Versions in the status string can be a lower priority follow-on enhancement to this follow-on enhancement. 😉

**Comment by dempseyatgithub** [comment]
I have an implementation of what we've been discussing available to try out at https://swift-test-site.xyz/swift-evolution.

**Comment by kaishin** [comment]
> And I'd like to use "In Active Review" in the status string since 'Active Review' is the only status that doesn't fit the sentence "This proposal is blank".

Good catch indeed!

> I have an implementation of what we've been discussing available to try out at https://swift-test-site.xyz/swift-evolution.

Lovely! I played around with it and the only nit I have is the "Edit" <-> "Hide" labels when no filters are selected. Personally I'd have gone with "Edit Filters" <-> "Done Editing Filters" or "Show Filters" <-> "Hide filters".

**Comment by dempseyatgithub** [comment]
> Lovely! I played around with it and the only nit I have is the "Edit" <-> "Hide" labels when no filters are selected. Personally I'd have gone with "Edit Filters" <-> "Done Editing Filters" or "Show Filters" <-> "Hide filters".

Thanks! I will update it to "Show Filters" <-> "Hide Filters" and see how that feels.

The only time you will see "Hide Filters" / "Done Editing Filters" is if you have no filters selected. For me, having no filters selected feels like you aren't 'done editing', you are just hiding the filters.

**Comment by kaishin** [comment]
> The only time you will see "Hide Filters" / "Done Editing Filters" is if you have no filters selected. For me, having no filters selected feels like you aren't 'done editing', you are just hiding the filters.

I also prefer Show/Hide as Edit filter is itself a bit inaccurate if you haven't added any filters yet.
